src/Operation/ReplaceOne.php

```diff
@@ -79,28 +83,21 @@ class ReplaceOne implements Executable
      */
     public function __construct(string $databaseName, string $collectionName, $filter, $replacement, array $options = [])
     {
-        if (! is_document($replacement)) {
-            throw InvalidArgumentException::expectedDocumentType('$replacement', $replacement);
-        }
-
-        // Treat empty arrays as replacement documents for BC
-        if ($replacement === []) {
-            $replacement = (object) $replacement;
+        if (isset($options['codec']) && ! $options['codec'] instanceof DocumentCodec) {
+            throw InvalidArgumentException::invalidType('"codec" option', $options['codec'], DocumentCodec::class);
         }
 
-        if (is_first_key_operator($replacement)) {
-            throw new InvalidArgumentException('First key in $replacement is an update operator');
-        }
-
-        if (is_pipeline($replacement, true /* allowEmpty */)) {
-            throw new InvalidArgumentException('$replacement is an update pipeline');
+        if (isset($options['codec'])) {
+            if (isset($options['typeMap'])) {
+                throw InvalidArgumentException::cannotCombineCodecAndTypeMap();
+            }
```

Comment:
```suggestion
        if (isset($options['codec'], $options['typeMap'])) {
            throw InvalidArgumentException::cannotCombineCodecAndTypeMap();
```

Alternatively: `isset() && isset()` if you think that's more readable.